library/core/src/iter/traits/double_ended.rs

```diff
@@ -294,11 +294,7 @@ pub trait DoubleEndedIterator: Iterator {
         Self: Sized,
         F: FnMut(B, Self::Item) -> B,
     {
-        let mut accum = init;
-        while let Some(x) = self.next_back() {
-            accum = f(accum, x);
-        }
-        accum
+        self.try_rfold(init, |accum, item| NeverShortCircuit(f(accum, item))).0
```

Comment:
Dunno if it'd impact perf, but there used to be a separate method for wrapping a function in `NeverShortCircuit`: https://github.com/rust-lang/rust/blob/0b2f717dfaf4835aa644d925a12c93db8f15dd61/library/core/src/ops/try_trait.rs#L463

Reply:
This is neat, although honestly I think that it's not *super* worth having this since the alternative is so short. I feel like the current code looks a bit clearer than:

```rust
self.try_rfold(init, NeverShortCircuit.wrap_mut_2(f))
```